staging/src/k8s.io/api/bulk/v1alpha1/types.go

```diff
@@ -0,0 +1,216 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +genclient:nonNamespaced
+
+// BulkRequest contains a single request to bulk api.
+type BulkRequest struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Opaque identifier used to match requests and responses.
+	// +optional
+	RequestID string `json:"requestId,omitempty" protobuf:"bytes,1,opt,name=requestId"`
+
+	// Starts new watch.
+	// +optional
+	Watch *WatchOperation `json:"watch,omitempty" protobuf:"bytes,2,opt,name=watch"`
+
+	// Starts new watch.
+	// +optional
+	WatchList *WatchListOperation `json:"watchList,omitempty" protobuf:"bytes,3,opt,name=watchList"`
+
+	// Stops existing watch.
+	// +optional
+	StopWatch *StopWatchOperation `json:"stopWatch,omitempty" protobuf:"bytes,4,opt,name=stopWatch"`
+
+	// Gets list of items.
+	// +optional
+	List *ListOperation `json:"list,omitempty" protobuf:"bytes,5,opt,name=list"`
```

Comment:
Similarly for List and Get here.

Reply:
Yes - that makes more sense in this case.

Reply:
Get is `LIST metadata.name=NAME`.  I don't see any reason to have Get by itself here.

Reply:
@smarterclayton - agree. But we treat the differently in main api (I mean GET and LIST are different calls with different set of options). Shouldn't we reflect it here too then?

Reply:
Also existing client libraries use different signatures for get/list operations.
https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/pod.go#L41